### ISSUE TITLE
Missing reference per token probability in advatage calculation

### ISSUE THREAD

**Issue by davidluciolu** [huggingface/open-r1#497]
In original GRPO objective, the advantage is divided with the reference per token probability, calculated as follows in formula (1) from DeepSeekMath:

![Image](https://github.com/user-attachments/assets/7aee349b-089a-488f-820c-abf6efcbf457)

However, $\pi_{old}$ is missing in trl's implementation of GRPO:
https://github.com/huggingface/trl/blob/fc2b041b58f6fbe766dceaec819bc5a8f9d209da/trl/trainer/grpo_trainer.py#L711
that
`per_token_loss = torch.exp(per_token_logps - per_token_logps.detach()) * advantages.unsqueeze(1)`.

If following the original GRPO objective, `per_token_loss` should be
```
per_token_loss = torch.exp(per_token_logps - per_token_logps.detach()) * advantages.unsqueeze(1) * torch.exp(-ref_per_token_logps)
```

Does this matters?

**Comment by davidluciolu** [comment]
Thanks! I get it.
